src/fsharp/service/ServiceAssemblyContent.fs

```diff
@@ -999,8 +999,11 @@ module ParsedInput =
 
         mkPos line ctx.Pos.Column
     
-    let tryFindNearestPointToInsertOpenDeclaration (currentLine: int) (ast: ParsedInput) (entity: Idents) (insertionPoint: OpenStatementInsertionPoint) =
+    let findNearestPointToInsertOpenDeclaration (currentLine: int) (ast: ParsedInput) (entity: Idents) (insertionPoint: OpenStatementInsertionPoint) =
         match tryFindNearestPointAndModules currentLine ast insertionPoint with
         | Some (scope, _, point), modules -> 
-            Some (findBestPositionToInsertOpenDeclaration modules scope point entity)
-        | _ -> None
\ No newline at end of file
+            findBestPositionToInsertOpenDeclaration modules scope point entity
+        | _ ->
+            // we failed to find insertion point because ast is empty for some reason, return top left point in this case  
```

Comment:
Do we want to do this anytime there is an empty ast? 
What if we have a comment at the top of the file with open declarations below it? Would that mean the new open declaration will go above the comment?

Reply:
Also, could we find out why the ast is empty?

Reply:
> What if we have a comment at the top of the file with open declarations below it? Would that mean the new open declaration will go above the comment?

Yes. Or maybe on top of the comment. Why?

> Also, could we find out why the ast is empty?

Why? We cannot find a position to insert the open statement, that's enough. Or you suggest to fix the parser?..

Reply:
@TIHan 

I looked at this, and it seems like the right fix to me, tryFindNearestPointAndModules is looking for the nearest open so that it can insert the new one below it, and finds none. 

In findNearestPointToInsertOpenDeclaration when no open was found it quite reasonably says.  Oh well, I couldn't find anywhere to insert the open, I guess I will use the top of the file.

It might be interesting to skip over leading comments, but that is a new feature and is not the intent of this PR.

Reply:
Skipping over comments would be not be trivial as we don't retain comments in the AST.

The reasoning why I mentioned it was because I thought the AST was giving us nothing even if we had open declarations, and we would just shove a open declaration at the top of the others, potentially skipping a comment if there was one there. However, I misinterpreted this. The case he is fixing is when there are no open declarations, which makes this fix absolutely fine.